esx_service/cli/vmdkops_post_update.11.1.py

```diff
@@ -0,0 +1,192 @@
+        - move all from old_uuid to STATIC_UUID
+        - symlinks "_DEFAULT" to STATIC_UUID
+    In single DB transcation
+        - replaces old_uuid with STATIC UUID in tenant_id field for all tables:
+          (privileges, vms,  tenants, volumes)
+    starts the service , and if all good removes backup DB
+
+    NOTE: this does not delete any data, so the Docker volumes will stay around
+          no matter if the code succeeds or fails
+    """
+
+    stop_service = True  # 'False' here makes debugging faster.
+
+    dbfile = auth_data.AUTH_DB_PATH
+
+    # STEP: check DB presense and fetch old_uuid
+    if not os.path.isfile(dbfile):
+        print("Config DB", dbfile, "is not found, nothing to update - exiting.")
+        sys.exit(0)
+
+    cursor = sqlite3.connect(dbfile).cursor()
+    cursor.execute("select * from tenants where name='{0}'".format(STATIC_NAME))
+    try:
+        tenant_id, tenant_name, tenant_desr, tenant_def_db = cursor.fetchone()
+    except TypeError:
+        print("Can't find '{0}' tenant, exiting".format(STATIC_NAME))
+        sys.exit(ERROR)
+
+    print("Found default tenant: {0} {1} {2} {3}".format(tenant_id,
+                                                         tenant_name, tenant_desr, tenant_def_db))
+
+    old_uuid = tenant_id
+    if old_uuid == STATIC_UUID:
+        print("*** DB seems to have been already migrated,  exiting ***")
+        sys.exit(OK)
+
+
+    # STEP: Stop the service and back up the DB
+    backup = dbfile + ".bck"
+    if os.path.isfile(backup):
+        print("Backup file '{0}' already exists - skipping DB backup".format(backup))
+    else:
+        print("Backing up Config DB to '{0}'".format(backup))
+        shutil.copy(dbfile, backup)
+
+    if stop_service:
+        print("Stopping vmdk-opsd service")
+        os.system("/etc/init.d/vmdk-opsd stop")
+
+    # STEP : convert dir names to new UUID if needed.
+    print("Starting conversion of _DEFAULT tenant directory names. old_uid is {0}".format(old_uuid))
+    stores = vmdk_utils.get_datastores()
+    if not stores:
+        print("Docker volume storage is not initialized - skipping directories patching")
+    else:
+        for ds_name, ds_id, ds_path in stores:
+            print("Working on Datastore '{0}'".format(ds_name))
+            # move stuff from old_uuid to new_uuid ()
+            new_dir = os.path.join(ds_path, STATIC_UUID)
+            old_dir = os.path.join(ds_path, old_uuid)
+            symlink_name = os.path.join(ds_path, STATIC_NAME)
+            print("  Info: ", new_dir, old_dir, symlink_name)
+            if not os.path.isdir(old_dir):
+                print("  Skipping {0} - not found".format(old_dir))
+                continue
+            if os.path.isdir(new_dir):
+                print("  Skipping {0} - {1} already exists".format(old_dir, new_dir))
+            print("  Renaming {0} to {1}".format(old_dir, new_dir))
+            os.rename(old_dir, new_dir)
```

Comment:
while executing this upgrade script on multi-esx testbed, line#131 is failing while invoking on second ESX host.

```
[root@sc-rdops-vm17-dhcp-15-100:~] python /usr/lib/vmware/vmdkops/bin/script.py 
Found default tenant: 5be05130-5e18-490b-ba99-42f4b798153d _DEFAULT This is a default tenant on 100 
Backing up Config DB to '/etc/vmware/vmdkops/auth-db.bck'
Stopping vmdk-opsd service
Stopping vmdkops-opsd with PID=1240898
vmdkops-opsd is not running
Starting conversion of _DEFAULT tenant directory names. old_uid is 5be05130-5e18-490b-ba99-42f4b798153d
Working on Datastore 'sharedVmfs-0'
  Info:  /vmfs/volumes/sharedVmfs-0/dockvols/11111111-1111-1111-1111-111111111111 /vmfs/volumes/sharedVmfs-0/dockvols/5be05130-5e18-490b-ba99-42f4b798153d /vmfs/volumes/sharedVmfs-0/dockvols/_DEFAULT
  Skipping /vmfs/volumes/sharedVmfs-0/dockvols/5be05130-5e18-490b-ba99-42f4b798153d - /vmfs/volumes/sharedVmfs-0/dockvols/11111111-1111-1111-1111-111111111111 already exists
  Renaming /vmfs/volumes/sharedVmfs-0/dockvols/5be05130-5e18-490b-ba99-42f4b798153d to /vmfs/volumes/sharedVmfs-0/dockvols/11111111-1111-1111-1111-111111111111
Traceback (most recent call last):
  File "/usr/lib/vmware/vmdkops/bin/script.py", line 187, in <module>
    main()
  File "/usr/lib/vmware/vmdkops/bin/script.py", line 129, in main
    os.rename(old_dir, new_dir)
OSError: [Errno 39] Directory not empty: '/vmfs/volumes/sharedVmfs-0/dockvols/5be05130-5e18-490b-ba99-42f4b798153d' -> '/vmfs/volumes/sharedVmfs-0/dockvols/11111111-1111-1111-1111-111111111111' ```